L1TriggerConfig/L1TConfigProducers/src/L1TCaloParamsOnlineProd.cc

```diff
@@ -210,23 +211,23 @@ bool L1TCaloParamsOnlineProd::readCaloLayer2OnlineSettings(l1t::CaloParamsHelper
 
 L1TCaloParamsOnlineProd::L1TCaloParamsOnlineProd(const edm::ParameterSet& iConfig)
     : L1ConfigOnlineProdBaseExt<L1TCaloParamsO2ORcd, l1t::CaloParams>(iConfig) {
+  m_setWhatProduced(iConfig).setConsumes(baseSettings_token);
   exclusiveLayer = iConfig.getParameter<uint32_t>("exclusiveLayer");
   transactionSafe = iConfig.getParameter<bool>("transactionSafe");
 }
 
 std::unique_ptr<const l1t::CaloParams> L1TCaloParamsOnlineProd::newObject(const std::string& objectKey,
                                                                           const L1TCaloParamsO2ORcd& record) {
   const L1TCaloParamsRcd& baseRcd = record.template getRecord<L1TCaloParamsRcd>();
-  edm::ESHandle<l1t::CaloParams> baseSettings;
-  baseRcd.get(baseSettings);
+  auto const& baseSettings = baseRcd.get(baseSettings_token);
 
   if (objectKey.empty()) {
     edm::LogError("L1-O2O: L1TCaloParamsOnlineProd") << "Key is empty";
     if (transactionSafe)
       throw std::runtime_error("SummaryForFunctionManager: Calo  | Faulty  | Empty objectKey");
     else {
       edm::LogError("L1-O2O: L1TCaloParamsOnlineProd") << "returning unmodified prototype of l1t::CaloParams";
-      return std::make_unique<const l1t::CaloParams>(*(baseSettings.product()));
+      return std::make_unique<const l1t::CaloParams>(baseSettings);
```

Comment:
If the intention is not to return a copy of `baseSettings`, there is a way to avoid that copy.

Reply:
Right, Again I implemented minimal substitution of the old code, can try and avoid non necessary copies

Reply:
I see, again I implemented a minimal update of the code, I can try avoiding unnecessary copies

Reply:
That's fine, especially given that the copy was there already. Here is an example how to properly pass a pointer to another ESProduct without additional copies
https://github.com/cms-sw/cmssw/blob/6f50dffe5ebd61297a487cc5e92568d7fd15658b/CalibTracker/SiPixelESProducers/plugins/SiPixelGenErrorDBObjectESProducer.cc#L82